### PR TITLE
[Feat] add key in remote signer

### PR DESCRIPTION
**Pull request by Marca23**
## Implementation details
- Add key to remote signer
- Add api-key header to remote signer request

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: cmd/keys.go [hunk 1]
```diff
@@ -129,12 +135,13 @@ $ %s k a eth test-key`, appName, appName)),
...
 					return err
 				}
-			} else if input.RemoteSigner.Address != "" && input.RemoteSigner.Url != "" {
+			} else if input.RemoteSigner.Address != "" && input.RemoteSigner.Url != "" && input.RemoteSigner.Key != "" {
 				key, err = app.AddRemoteSignerKey(
 					chainName,
 					keyName,
 					input.RemoteSigner.Address,
 					input.RemoteSigner.Url,
+					input.RemoteSigner.Key,
```
[the remark this answers is not in the document]
Comment (tanut32039) [reply cmd/keys.go:144]: yes, should be optional

File: cmd/keys.go [hunk 2]
```diff
@@ -284,25 +292,28 @@ $ %s k s eth test-key`, appName, appName)),
...
 	// if a mnemonic is provided, no other input should be present
 	if input.Mnemonic != "" &&
-		(input.PrivateKey != "" || input.RemoteSigner.Address != "" || input.RemoteSigner.Url != "") {
+		(input.PrivateKey != "" || input.RemoteSigner.Address != "" || input.RemoteSigner.Url != "" || input.RemoteSigner.Key != "") {
 		return fmt.Errorf("mnemonic cannot be provided with private key or remote signer")
 	}
 
 	// if any remote-signer field is provided, it must be the only input
-	if input.RemoteSigner.Address != "" || input.RemoteSigner.Url != "" {
+	if input.RemoteSigner.Address != "" || input.RemoteSigner.Url != "" || input.RemoteSigner.Key != "" {
```
Comment (tanut32039) [inline cmd/keys.go:306]: `input.RemoteSigner.Key ` can be optional

File: relayer/wallet/geth/remote_signer.go [hunk 3]
```diff
@@ -55,8 +58,10 @@ func (r *RemoteSigner) GetAddress() (addr string) {
 
 // Sign requests the remote KMS to sign the data and returns the signature.
 func (r *RemoteSigner) Sign(data []byte) ([]byte, error) {
+	md := metadata.Pairs("api-key", r.Key)
+	ctx := metadata.NewOutgoingContext(context.Background(), md)
```
Comment (tanut32039) [inline relayer/wallet/geth/remote_signer.go:62]: since api-key can be optional, ctx can be only context.Background() when that key has does not required api-key

File: relayer/chains/evm/keys_test.go [hunk 4]
```diff
@@ -181,10 +181,12 @@ func (s *KeysTestSuite) TestAddKeyByMnemonic() {
 }
 
 func (s *KeysTestSuite) TestAddRemoteSignerKey() {
```
Comment (RogerKSI) [inline relayer/chains/evm/keys_test.go:183]: add test for null key as well.

**Review by tanut32039 — APPROVED** [review]
